cli/internal/cmd/upgradecheck.go

```diff
@@ -576,7 +576,7 @@ func (v *versionUpgrade) writeConfig(conf *config.Config, fileHandler file.Handl
 		conf.MicroserviceVersion = v.newServices
 	}
 	if len(v.newKubernetes) > 0 {
-		conf.KubernetesVersion = v.newKubernetes[0]
+		conf.KubernetesVersion = versions.ValidK8sVersion(v.newKubernetes[0])
```

Comment:
```suggestion
		conf.KubernetesVersion = versions.NewValidK8sVersion(v.newKubernetes[0], true)
```
I know, this should be safe. But better safe than sorry. And if others see this use of the type they may start imitating it.